demo/README.md

```diff
@@ -0,0 +1,203 @@
+as the client who verifies the final product.
+
+
+### Download and setup in-toto on *NIX (Linux, OS X, ..)
+```shell
+# Make sure you have git, python and pip installed on your system
+# and get in-toto
+git clone -b develop --recursive https://github.com/in-toto/in-toto.git
+
+# Change into project root directory
+cd in-toto
+
+# Install required packages (we strongly recommend using Virtual Environments)
+# http://docs.python-guide.org/en/latest/dev/virtualenvs/
+pip install -r requirements.txt
+
+# Install additional requirements that for some good reason are not in the
+# requirements file
+pip install pycrypto cryptography
+
+# Make sure in-toto is on your PYTHONPATH and used commands are on your PATH
+export PYTHONPATH=$PYTHONPATH:$PWD
+export PATH=$PATH:$PWD/toto
+
+# Change into the demo directoy and you are ready to start
+cd demo
+```
+Inside the demo directory you will find four directories: `owner_alice`,
+`functionary_bob`, `functionary_carl` and `final_product`. Alice, Bob and Carl
+already have RSA keys in each of their directories. This is what you see:
+```shell
+tree
+# the tree command gives you the following output
+# ├── README.md
+# ├── final_product
+# ├── functionary_bob
+# │   ├── bob
+# │   └── bob.pub
+# ├── functionary_carl
+# │   ├── carl
+# │   └── carl.pub
+# ├── owner_alice
+# │   ├── alice
+# │   └── alice.pub
+# └── run-demo.py
+```
+
+### Define supply chain layout (Alice)
+First, we will need to define the supply chain layout. To simplify this process,
+we provide a script that generates a simple layout for the purpose of the demo.
+In this supply chain layout, we have Alice, who is the project owner that creates
+the layout, Bob, who uses `vi` to create a Python program foo.py, and Carl, who
+uses `tar` to package up `foo.py` into a tarball which together with the in-toto
+metadata composes the final product that will eventually be installed and verified
+by the end user.
+
+```shell
+# Create and sign the supply chain layout on behalf of Alice
+cd owner_alice
+python create_layout.py
+```
+The script will create a layout, add Bob's and Carl's public keys (fetched from
+their directories), sign it with Alice's private key and dump it to `root.layout`.
+In `root.layout`, you will find that (besides the signature and other information)
+there are two steps, `write_code` and `package`, that the functionaries Bob
+and Carl, identified by their public keys, are authorized to perform.
+
+### Write code (Bob)
+Now, we will take the role of the functionary Bob and perform the step
+`write-code` on his behalf, that is we use in-toto to open an editor and record
+metadata for what we do. Execute the following commands to change to Bob's
+directory and perform the step.
+
+```shell
+cd ../functionary_bob
+toto-run.py --step-name write-code --products foo.py --key bob -- vi foo.py
+# Bob has to send the resulting foo.py to Carl so that he can package it
+cp foo.py ../functionary_carl/
```

Comment:
Or put a blank line between the two commands:

```Python
cd ../functionary_bob
toto-run.py --step-name write-code --products foo.py --key bob -- vi foo.py

# Bob has to send the resulting foo.py to Carl so that he can package it
cp foo.py ../functionary_carl/
```